### PR TITLE
Vala cleanup

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Closes: #32800

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libskk/default.nix [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-    rev = "6a232e75de6d5dbe543ab17c9b85dc7560093509";
-    sha256 = "1xa9akf95jyi4laiw1llnjdpfq5skhidm7dnkd0i0ds6npzzqnxc";
+    rev = version;
+    sha256 = "092bjir866f350s4prq9q0yg34s91vmr8wbgf2vh3kcax1yj1axm";
   };
 
   buildInputs = [ skk-dicts ];
-  nativeBuildInputs = [ vala gnome_common gobjectIntrospection libtool intltool pkgconfig ];
-  propagatedBuildInputs = [ libgee_0_8 json_glib ];
+  nativeBuildInputs = [ vala gnome_common gobjectIntrospection libtool gettext pkgconfig ];
```
Comment (Mic92) [inline pkgs/development/libraries/libskk/default.nix]: just out of interest, when do I use `gettext`, when do I want `intltool`?

File: pkgs/development/compilers/vala/default.nix [hunk 2]
```diff
@@ -59,6 +53,12 @@ in rec {
     sha256  = "16cjybjw100qps6jg0jdyjh8hndz8a876zmxpybnf30a8vygrk7m";
   };
 
+  vala_0_36 = generic {
```
Comment (Mic92) [inline pkgs/development/compilers/vala/default.nix]: What depends on `vala_0_36`?

File: pkgs/development/libraries/libskk/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/libraries/libskk/default.nix]: `intltool` is basically abandoned now and `gettext` should be able to do all the things you needed `intltool` for. Though the helper commands are different so you can only drop intltools when a package does so (reading changelogs is useful). See also https://wiki.gnome.org/Initiatives/GnomeGoals/GettextMigration

File: pkgs/development/compilers/vala/default.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/development/compilers/vala/default.nix]: Hmm, you are right, I must have confused it with 0.26.

**Comment by jtojnar** [comment]
Okay, now we have 

* `vala_0_26`:
    * [`finalterm`](https://github.com/p-e-w/finalterm) abandoned
* `vala_0_28`:
    * [`vanubi`](https://github.com/vanubi/vanubi) @lethalman do you still use it
* `vala_0_34`:
    * [`midori`](http://midori-browser.org/news/)
    * [`synapse`](https://bugs.launchpad.net/synapse-project/) https://bugs.launchpad.net/synapse-project/+bug/1738964

but I am not sure if it is worse to keep them all on the common supported version and drop the rest, or keep each on its own most recent supported version leaving the old valas in the repo.

**Comment by Mic92** [comment]
Looks good to me for the moment. The evaluation error should be fixed though.

**Comment by grahamc** [comment]
@GrahamcOfBorg eval

**Comment by jtojnar** [comment]
Oh, I forgot to commit the finalterm fix, it is okay now.

File: pkgs/development/web/valum/default.nix [hunk 3]
```diff
@@ -1,29 +1,24 @@
...
-  buildPhase = ''python waf build'';
-
-  installPhase = ''python waf install'';
+  nativeBuildInputs = [ meson ninja pkgconfig ];
+  buildInputs = [ glib vala ctpl libgee libsoup fcgi ];
 
   meta = with stdenv.lib; {
     homepage = https://github.com/valum-framework/valum;
     description = "Web micro-framework written in Vala";
+    license = licenses.lgpl;
```
Comment (orivej) [inline pkgs/development/web/valum/default.nix]: There is no such attribute.

Comment (jtojnar) [reply pkgs/development/web/valum/default.nix]: Oh, right. Is there a way, I can check metadata locally?

Comment (jtojnar) [reply pkgs/development/web/valum/default.nix]: Turns out I can just run `nix-build nixos/release.nix`

Comment (orivej) [reply pkgs/development/web/valum/default.nix]: https://github.com/NixOS/nixpkgs/commit/a9d212fea4d6631052d143d7151cacad909beabe#commitcomment-26368327

**Comment by jtojnar** [comment]
@GrahamcOfBorg build zssh libosinfo osinfo-db osinfo-db-tools

**Comment by orivej** [comment]
Tested with nox-review and merged.

**Pull request merged**